### PR TITLE
(WIP) Get Nvblox to work with Gazebo simulation (NOT READY YET)

### LINKED ISSUE #54: Bringup sensors in Gazebo so we can use nvblox & nav2 with Gazebo for simulating autonomy
I don't think that Zed is supported in Gazebo, but Jonathan mentioned that we could use a lidar + camera with nvblox instead. We will need to use the camera for detecting Apriltags as well, which should be placed in the correct spots on the field.

### PR DESCRIPTION
**Pull request by Isopod00**
Get Nvblox to work with our Gazebo simulation so we can do more testing in simulation!

### CODE REVIEW

File: src/gazebo/ros_gz_description/models/master_ASM/urdf/master_ASM.urdf [hunk 1]
```diff
@@ -76,23 +76,52 @@
...
           </camera>
           <topic>camera</topic>
       </sensor>
+
+      <sensor name="depthcamera" type="depth">
+          <!-- <visualize>true</visualize> -->
+          <always_on>true</always_on>
+          <update_rate>30</update_rate>
+          <camera>
+        <horizontal_fov>1.1</horizontal_fov> # TODO: This should be 1.91986218 but it doesn't seem to work with a value that high
```
Comment (Isopod00) [inline src/gazebo/ros_gz_description/models/master_ASM/urdf/master_ASM.urdf]: idk what is causing this (horizontal_fov issue) but we should try to fix it

**Comment by jmblixt3** [comment]
Idk how i closed this

**Comment by jmblixt3** [comment]
Reopened

**Comment by jmblixt3** [comment]
Reopened

File: src/gazebo/ros_gz_description/models/master_ASM/urdf/master_ASM.urdf (hunk 1 above)
Comment (Isopod00) [reply src/gazebo/ros_gz_description/models/master_ASM/urdf/master_ASM.urdf]: I think this is because when it sees over the arena walls it sees infinite depth or something. @pwmarshal, can you please make the arena walls taller for the UCF field?

File: config/rviz/zed_example.rviz [hunk 2]
```diff
@@ -44,7 +44,7 @@ Visualization Manager:
         zed2i_camera_center:
           Value: false
         zed2i_camera_link:
-          Value: true
```
Comment (Isopod00) [inline config/rviz/zed_example.rviz]: we should undo this change

File: src/gazebo/gazebo_files/worlds/NASA_field.sdf [hunk 3]
```diff
@@ -31,8 +35,17 @@
     </include>
 
     <include>
-      <uri>package://robot_description/models/master_ASM</uri>
-      <pose>-1.8 -1.5 1 0 0.52359878 0</pose>
+      <uri>package://ros_gz_description/models/master_ASM</uri>
+      <pose>-2.44 -1.5 1 0 0.52359878 0</pose>
```
Comment (Isopod00) [inline src/gazebo/gazebo_files/worlds/NASA_field.sdf]: move this position so the robot spawns in a good spot

File: src/gazebo/ros_gz_launch/config/camera.rviz [hunk 4]
```diff
@@ -26,23 +26,28 @@ Panels:
...
-        Z: 0
-      Plane: XY
-      Plane Cell Count: 10
-      Reference Frame: <Fixed Frame>
+      Frame Timeout: 15
+      Frames:
+        All Enabled: true
+        base_link:
+          Value: true
+        map:
```
Comment (Isopod00) [inline src/gazebo/ros_gz_launch/config/camera.rviz]: Should we add more frames to this like maybe the zed2i camera link frame?

File: src/isaac_ros/isaac_ros_launch/package.xml [hunk 5]
```diff
@@ -24,4 +24,4 @@
   <export>
     <build_type>ament_python</build_type>
   </export>
-</package>
+</package>
```
Comment (Isopod00) [inline src/isaac_ros/isaac_ros_launch/package.xml]: Revert this change because it isn't changing anything

File: src/robot_description/models/master_ASM/urdf/master_ASM.urdf.xacro [hunk 6]
```diff
@@ -49,30 +49,59 @@ Stephen Brawner (brawner@gmail.com)
   <xacro:zed_camera name="zed2i" model="zed2i">
   </xacro:zed_camera>
   <joint name="camera_joint" type="fixed">
-    <child link="base_link"/>
-    <parent link="zed2i_camera_link"/>
```
Comment (Isopod00) [inline src/robot_description/models/master_ASM/urdf/master_ASM.urdf.xacro]: revert this change here? it's not changing anything

**Comment by jmblixt3** [comment]
@Isopod00 can we close this in favor of #221

**Comment by jmblixt3** [comment]
Closed in favor of #221
